src/setupWorker/glossary.ts

```diff
@@ -54,6 +54,14 @@ export type StartOptions = SharedOptions & {
    * instance is ready. Defaults to `true`.
    */
   waitUntilReady?: boolean
+
+  /**
+   * A boolean that allows you to override the default strict matching behavior. If set to `true`, the first worker
+   * based on a matching filename only (ex: mockServiceWorker.js) will be returned. This is primarily useful in the scenario that
+   * you are using proxies and having difficulty to get the origin and `scriptURL` to match due to port differences.
+   * Defaults to `false`.
+   */
+  matchFilenameOnly?: boolean
```

Comment:
I read the issue and I think this solves that use case (and probably others too).
Another option would be to give the user this responsibility, which would cover all use cases.
It would be more complicated for the user, but it would be more flexible.

I'm not sure it needs to be more flexible tho. Perhaps this is something to keep in mind for when another use case pops up, before adding other options.

e.g.

```ts
serviceWorkerFileMatcher?: (scriptUrl:string) => boolean

if (serviceWorkerFileMatcher) {
  return serviceWorkerFileMatcher(worker.scriptURL)
}
return worker.scriptURL === absoluteWorkerUrl
```

Reply:
Hey, thanks for this idea! I'm a big fan of matcher funcs as well _(in this instance, as well as basically any other time it's relevant)_. I think the only real distinction is that what I'm proposing is the viewpoint: 'as a library, we've thought of what you're going to run into, and this is the easiest way to solve it', and the flag becomes obvious. The matcher func requires slightly more 🧠 power and documentation, but I'd agree that it is probably the better long-term solution. 

My only challenge to the matcher func is basically what you said: "not sure it needs to be more flexible". That being said, I'm more than happy to change this to your suggestion and add in the tests once we have a solid decision 🎉.

Reply:
I agree that we should lean towards expanding API outwards (delegating more control to the user), not inwards (encapsulating specific behaviors under flags). Thanks for pointing this out, @timdeschryver!

Depending on when such matcher would be called, we can expose more information about the worker. Ideally, a `ServiceWorkerRegistration` reference, which contains about all you may need to create a custom registration predicate.